pkg/admission/no_downscale.go

```diff
@@ -140,6 +140,12 @@ func allowErr(logger log.Logger, msg string, err error) *v1.AdmissionResponse {
 }
 
 func getResourceLabels(ctx context.Context, ar v1.AdmissionReview, api kubernetes.Interface) (map[string]string, error) {
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	default:
+	}
```

Comment:
Also, this is shorter and equivalent:
```suggestion
	if err := ctx.Err(); err != nil {
		return nil, err
	}
```